### ISSUE TITLE
Union type safety

### ISSUE THREAD

**Issue by heinrichreimer** [prisma/graphqlgen#346]
## Description
graphqlgen doesn't enforce type safety for union types.

## Steps to reproduce
Schema: 
```graphql
union Availability = EstimatedAvailability | RemainingAvailability
type EstimatedAvailability {
    timestamp: Int!
}
type RemainingAvailability {
    remainingItems: Int!
}
type Offer {
    availability: Availability!
}
```
Model classes:
```typescript
export type Availability = EstimatedAvailability | RemainingAvailability
export interface EstimatedAvailability {
	timestamp: number
}
export interface RemainingAvailability {
	remainingItems: number
}
export interface Offer {
	availability: Availability
}
```

## Expected results
```typescript
export namespace OfferResolvers {

  export type AvailabilityResolver = (
    parent: Offer,
    args: {},
    ctx: Context,
    info: GraphQLResolveInfo
  ) => Availability | Promise<Availability>;

  export interface Type {
    availability: (
      parent: Offer,
      args: {},
      ctx: Context,
      info: GraphQLResolveInfo
    ) => Availability | Promise<Availability>;
  }
}
```

## Actual results
```typescript
export namespace OfferResolvers {

  export type AvailabilityResolver = (
    parent: Offer,
    args: {},
    ctx: Context,
    info: GraphQLResolveInfo
  ) => {} | Promise<{}>;

  export interface Type {
    availability: (
      parent: Offer,
      args: {},
      ctx: Context,
      info: GraphQLResolveInfo
    ) => {} | Promise<{}>;
  }
}
```

## Versions

- graphqlgen: 0.4.0
- OS name and version: Node 8.11.2, NPM 6.4.1, Windows 10, 64 Bit

**Comment by timsuchanek** [comment]
This is related to https://github.com/prisma/graphqlgen/pull/149

**Comment by jasonkuhrt** [comment]
Resolved by eb2f3cf

**Comment by jasonkuhrt** [comment]
Results now:

```ts
export namespace OfferResolvers {
  export const defaultResolvers = {
    availability: (parent: Offer) => parent.availability
  };

  export type AvailabilityResolver = (
    parent: Offer,
    args: {},
    ctx: Context,
    info: GraphQLResolveInfo
  ) =>
    | EstimatedAvailability
    | RemainingAvailability
    | null
    | Promise<EstimatedAvailability | RemainingAvailability | null>;

  export interface Type {
    availability: (
      parent: Offer,
      args: {},
      ctx: Context,
      info: GraphQLResolveInfo
    ) =>
      | EstimatedAvailability
      | RemainingAvailability
      | null
      | Promise<EstimatedAvailability | RemainingAvailability | null>;
  }
}
```
